internal/test/crds.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/pointer"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+var (
+	bootstrapGroupVersion = schema.GroupVersion{Group: "bootstrap.cluster.x-k8s.io", Version: "v1alpha4"}
+
+	// BootstrapCRD is a generic boostrap CRD.
+	BootstrapCRD = generateCRD(bootstrapGroupVersion.WithKind("Bootstrap"))
```

Comment:
I renamed BootstrapMachine to Bootstrap and BootstrapMachineTemplate to BootstrapTemplate as that are the names with which we usually refer to them. This hat the downside that I had to adjust occurrences of those kinds in the unit tests, but I think that's worth it.

Reply:
if we are renaming let's use `GenericBootstrapConfig`, which is consistent with existing names (e.g `KubeadmBootstrapConfig`)
PTAL names used in clusterctl crds.

Reply:
```suggestion
	GenericBootstrapCRD = generateCRD(bootstrapGroupVersion.WithKind("Bootstrap"))
```
Let's keep generic as a prefix